src/app/address/AddressFormModal.tsx

```diff
@@ -0,0 +1,116 @@
+import { Country, FormField } from '@bigcommerce/checkout-sdk';
+import { withFormik, FormikProps } from 'formik';
+import React, { FunctionComponent } from 'react';
+import { lazy } from 'yup';
+
+import { preventDefault } from '../common/dom';
+import { withLanguage, TranslatedString, WithLanguageProps } from '../locale';
+import { Button, ButtonVariant } from '../ui/button';
+import { Form } from '../ui/form';
+import { LoadingOverlay } from '../ui/loading';
+import { Modal, ModalHeader } from '../ui/modal';
+
+import getAddressFormFieldsValidationSchema from './getAddressFormFieldsValidationSchema';
+import { AddressFormValues } from './mapAddressToFormValues';
+import AddressForm from './AddressForm';
+
+export interface AddressFormModalProps {
+    countries?: Country[];
+    countriesWithAutocomplete: string[];
+    googleMapsApiKey?: string;
+    isLoading: boolean;
+    shouldShowSaveAddress?: boolean;
+    isOpen: boolean;
+    defaultCountryCode?: string;
+    getFields(countryCode?: string): FormField[];
+    onSaveAddress(address: AddressFormValues): void;
+    onRequestClose?(): void;
+    onAfterOpen?(): void;
+}
+
+const AddressModal: FunctionComponent<AddressFormModalProps & WithLanguageProps & FormikProps<AddressFormValues>> = ({
```

Comment:
It'd be nice if the name of the component is the same as the file (`AddressFormModal`).